ec/ecresource/deploymentresource/expanders.go

```diff
@@ -60,22 +60,25 @@ func createResourceToModel(d *schema.ResourceData, client *api.API) (*models.Dep
 	}
 	result.Resources.Kibana = append(result.Resources.Kibana, kibanaRes...)
 
-	apmRes, err := expandApmResources(
-		d.Get("apm").([]interface{}), apmResource(res),
-	)
-	if err != nil {
-		return nil, err
+	if res.DeploymentTemplate.Resources.Apm != nil {
+		apmRes, err := expandApmResources(
+			d.Get("apm").([]interface{}), apmResource(res),
+		)
+		if err != nil {
+			return nil, err
+		}
+		result.Resources.Apm = append(result.Resources.Apm, apmRes...)
+	}
```

Comment:
I'd rather guard a single line with an `if` clause than all the lines above.

```suggestion
	apmRes, err := expandApmResources(
		d.Get("apm").([]interface{}), apmResource(res),
	)
	if err != nil {
		return nil, err
	}
	if len(apmRes) > 0 {
		result.Resources.Apm = append(result.Resources.Apm, apmRes...)
	}
```

Reply:
this might now work because the error is thrown from inside `apmResource` function

Reply:
Have updated #129  with the error message